### ISSUE TITLE
Regexp parsing error: got incomplete JSON array configuration at

### ISSUE THREAD

**Issue by bkupidura** [fluent/fluent-plugin-rewrite-tag-filter#58]
When rule pattern starts with `[`(?) td-agent fails on start with json parsing error:

```
/opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/basic_parser.rb:92:in `parse_error!': got incomplete JSON array configuration at label-systemd.conf line 21,9 (Fluent::ConfigParseError)
 20: </match>
 21: </label>

     ---------^
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/literal_parser.rb:246:in `scan_json'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/literal_parser.rb:60:in `parse_literal'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:130:in `parse_element'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:95:in `parse_element'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:95:in `parse_element'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:95:in `parse_element'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:166:in `block in eval_include'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:160:in `each'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:160:in `eval_include'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:145:in `parse_include'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:104:in `parse_element'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:43:in `parse!'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config/v1_parser.rb:33:in `parse'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/config.rb:39:in `parse'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/supervisor.rb:741:in `read_config'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/supervisor.rb:451:in `run_supervisor'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/lib/fluent/command/fluentd.rb:310:in `<top (required)>'
        from /opt/td-agent/embedded/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/td-agent/embedded/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /opt/td-agent/embedded/lib/ruby/gems/2.4.0/gems/fluentd-1.0.2/bin/fluentd:8:in `<top (required)>'
        from /opt/td-agent/embedded/bin/fluentd:22:in `load'
        from /opt/td-agent/embedded/bin/fluentd:22:in `<top (required)>'
        from /usr/sbin/td-agent:7:in `load'
        from /usr/sbin/td-agent:7:in `<main>'
```

Config:
```
<source>
  @type forward
  @label @test
  tag test
  port 24224
  bind 0.0.0.0
</source>

<label @test>
<match test>
  @type rewrite_tag_filter
  <rule>
    key Payload
    pattern [sv]d[a-z]+\d*
    tag metric.hdd_errors
  </rule>
</match>
<match metric.**>
  @type stdout
</match>
</label>
```

If you put anything before `[sv]` fluentd will start ...

td-agent 1.0.2
fluent-plugin-rewrite-tag-filter 2.0.1

**Comment by y-ken** [comment]
Hi @bkupidura 
You can try `pattern '[sv]d[a-z]+\d*'` instead of `pattern [sv]d[a-z]+\d*` at the rule.

**Comment by bkupidura** [comment]
Yep, this helps. But other *regexp characters* (`^`) works as expected. Maybe we can update readme to put `'` in examples.

**Comment by y-ken** [comment]
Thank you for your advice. I'm considering the what is the best solution.

**Comment by okkez** [comment]
Released v2.1.0.
